rfcs/45-stageless.md

```diff
@@ -0,0 +1,761 @@
+# Feature Name: `stageless`
+
+## Summary
+
+Bevy's current scheduling architecture (as of 0.7) has a large number of issues due to internal complexity.
+Certain concepts and behaviors are unnecessarily fused together and certain building blocks—stages, run criteria, states—are presented as independent but actually have significant internal coupling.
+These properties frequently lead users into encounters with unexpected side effects and indecipherable errors.
+
+It's become a tangled mess.

```

Comment:
nit: this isn't really necessary to mention here as the prior paragraph already describes the trouble users have when working with scheduling.